File: frontend/src/modules/sales/types/sale.ts
export type VentaEstado = 'pendiente' | 'completada' | 'anulada'

export type MetodoPago = 'efectivo' | 'transferencia' | 'tarjeta' | 'otro'

export interface VentaDetalle {
  id: number
  venta_id: number
  producto_id: number
  cantidad: number
  precio_unitario: number
  descuento: number
  subtotal: number
  created_at: string
  updated_at: string
  producto_nombre?: string
  producto_sku?: string | null
  producto_imagen?: string | null
}

export interface Venta {
  id: number
  numero: string
  usuario_id: number
  cliente_id: number | null
  fecha_venta: string
  subtotal: number
  descuento: number
  total: number
  metodo_pago: MetodoPago
  estado: VentaEstado
  created_at: string
  updated_at: string
  detalles: VentaDetalle[]
  cliente_nombre?: string
  usuario_nombre?: string
}

export interface VentaFilter {
  estado?: VentaEstado
  fecha_inicio?: string
  fecha_fin?: string
  usuario_id?: number
  q?: string
}

export interface VentaListResponse {
  items: Venta[]
  total: number
}

export interface VentaDetalleCreate {
  producto_id: number
  cantidad: number
  precio_unitario: number
  descuento: number
}

export interface VentaCreate {
  usuario_id: number
  cliente_id: number | null
  metodo_pago: MetodoPago
  estado: 'completada'
  descuento: number
  detalles: VentaDetalleCreate[]
}

export interface VentaResponse {
  id: number
  numero: string
  usuario_id: number
  cliente_id: number | null
  fecha_venta: string
  subtotal: number
  descuento: number
  total: number
  metodo_pago: string
  estado: string
  created_at: string
  updated_at: string
  detalles: VentaDetalleResponse[]
}

export interface VentaDetalleResponse {
  id: number
  venta_id: number
  producto_id: number
  cantidad: number
  precio_unitario: number
  descuento: number
  subtotal: number
  created_at: string
  updated_at: string
}

export const ESTADOS_INMODIFICABLES: VentaEstado[] = ['anulada']

File: backend/app/schemas/sale.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, ConfigDict


class VentaEstado(str, Enum):
    pendiente = 'pendiente'
    completada = 'completada'
    anulada = 'anulada'


class MetodoPago(str, Enum):
    efectivo = 'efectivo'
    transferencia = 'transferencia'
    tarjeta = 'tarjeta'
    otro = 'otro'


class VentaDetalleCreate(BaseModel):
    producto_id: int
    cantidad: int
    precio_unitario: int
    descuento: int = 0


class VentaDetalleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    venta_id: int
    producto_id: int
    cantidad: int
    precio_unitario: int
    descuento: int
    subtotal: int
    created_at: datetime
    updated_at: datetime


class VentaCreate(BaseModel):
    usuario_id: int
    cliente_id: int | None = None
    metodo_pago: MetodoPago
    estado: VentaEstado = VentaEstado.completada
    descuento: int = 0
    detalles: list[VentaDetalleCreate]


class VentaUpdate(BaseModel):
    estado: VentaEstado | None = None


class VentaListResponse(BaseModel):
    items: list[VentaResponse]
    total: int


class VentaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    numero: str
    usuario_id: int
    cliente_id: int | None
    cliente_nombre: str | None = None
    fecha_venta: datetime
    subtotal: int
    descuento: int
    total: int
    metodo_pago: MetodoPago
    estado: VentaEstado
    created_at: datetime
    updated_at: datetime
    detalles: list[VentaDetalleResponse] = []
